### PR TITLE
feat(vpcgw): add support for vpc public gateway

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: scaleway/resource_vpc_public_gateway.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+				Optional:    true,
+				Description: "Description of a public gateway",
+			},
+			"type": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The type of the gateway",
+				DiffSuppressFunc: diffSuppressFuncIgnoreCase,
+			},
+			"upstream_dns_servers": {
```
Comment (remyleone) [inline scaleway/resource_vpc_public_gateway.go:40]: Add tests for setting upstream dns servers

File: scaleway/resource_vpc_public_gateway.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "override the gateway's default recursive DNS servers, if DNS features are enabled",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"ip_id": {
```
Comment (remyleone) [inline scaleway/resource_vpc_public_gateway.go:48]: add support for ip id. Is it compatible with other IP from instance ? From LB ?

File: scaleway/resource_vpc_public_gateway.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+func resourceScalewayVPCPublicGatewayCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcgwAPI, zone, err := vpcgwAPIWithZone(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	req := &vpcgw.CreateGatewayRequest{
+		Name: expandOrGenerateString(d.Get("name"), "pn"),
+		// TODO: Add the description field
```
Comment (remyleone) [inline scaleway/resource_vpc_public_gateway.go:88]: Description field is missing from the SDK

File: scaleway/resource_vpc_public_gateway_test.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+				Config: fmt.Sprintf(`
+					resource scaleway_vpc_public_gateway main {
+						name = "%s-new"
+						type = "VPC-GW-S"
+						tags = ["tag0", "tag1"]
+						upstream_dns_servers = [ "1.2.3.4", "4.3.2.1" ]
+					}
+				`, publicGatewayName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayVPCPublicGatewayExists(tt, "scaleway_vpc_public_gateway.main"),
```
Comment (jeansebastienh) [inline scaleway/resource_vpc_public_gateway_test.go:83]: Could you test the new name ?

**Review by jeansebastienh — APPROVED** [review]

**Pull request merged**